### PR TITLE
Add secure to site

### PR DESCRIPTION
**Pull request by PetrushynskyiOleksii**
**Ref #23 (2)**

### CODE REVIEW

File: mysite/urls.py [hunk 1]
```diff
@@ -3,7 +3,11 @@
 from django.conf.urls import url, include
 from django.contrib import admin
 
+from django.contrib.auth import views
+
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
+    url(r'^accounts/login/$', views.login, name='login'),
+    url(r'^accounts/logout/$', views.logout, name='logout', kwargs={'next_page': '/'}),
```
Comment (webknjaz) [inline mysite/urls.py]: why not read the next page from settings?

File: blog/tests/test_views.py [hunk 2]
```diff
@@ -110,23 +118,30 @@ def test_post_edit(self):
...
         self.assertTrue(authorization)
         response = self.client.get(reverse('post_draft_list'))
         self.assertEqual(200, response.status_code)
 
     def test_publish_post(self):
         """Test for publishing post."""
-        response = self.client.get(reverse('post_publish', kwargs={'pk': 9999}))
-        self.assertEqual(404, response.status_code)
+        response = self.client.get(reverse('post_publish', kwargs={'pk': 1}))
+        self.assertEqual(302, response.status_code)
```
Comment (webknjaz) [inline blog/tests/test_views.py:131]: Please check where you've got redirected.

File: blog/tests/test_views.py [hunk 3]
```diff
@@ -110,23 +118,30 @@ def test_post_edit(self):
...
         authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
         self.assertTrue(authorization)
+        response = self.client.get(reverse('post_publish', kwargs={'pk': 9999}))
```
Comment (webknjaz) [inline blog/tests/test_views.py:134]: Why does post creation even accept kwarg? 0_O

File: blog/views.py [hunk 4]
```diff
@@ -15,10 +16,14 @@ def post_list(request):
 
 def post_detail(request, pk):
     """Render page with details of specific post."""
-    post = get_object_or_404(Post, pk=pk)
-    return render(request, 'blog/post_detail.html', {'post': post})
+    if request.user.is_authenticated:
+        post = get_object_or_404(Post, pk=pk)
+        return render(request, 'blog/post_detail.html', {'post': post})
+    else:
+        return redirect('login')
```
Comment (webknjaz) [inline blog/views.py:23]: don't you allow guests see the content?

File: blog/tests/test_views.py [hunk 5]
```diff
@@ -110,23 +118,30 @@ def test_post_edit(self):
...
+        self.assertEqual(404, response.status_code)
         post = Post.objects.create(id=9999, author=self.user, title='Test Publish', text='Text Publish')
         response = self.client.get(reverse('post_publish', kwargs={'pk': post.pk}), follow=True)
         self.assertRedirects(response, reverse('post_detail', kwargs={'pk': post.pk}))
 
     def test_delete_post(self):
         """Test for deleting post."""
+        response = self.client.get(reverse('post_remove', kwargs={'pk': self.first_post.pk}))
+        self.assertEqual(302, response.status_code)
```
Comment (webknjaz) [inline blog/tests/test_views.py:143]: where?

File: blog/tests/test_views.py [hunk 6]
```diff
@@ -93,6 +99,8 @@ def test_post_new_view(self):
 
     def test_post_edit(self):
         """Test view for edit post."""
+        response = self.client.get(reverse('post_edit', kwargs={'pk': self.first_post.pk}))
+        self.assertEqual(302, response.status_code)
```
Comment (webknjaz) [inline blog/tests/test_views.py:103]: where does it redirect the user? forbidden action must result in 401 or 403 depending on whether one's been authenticated or not.

File: mysite/urls.py (hunk 1 above)
Comment (PetrushynskyiOleksii) [reply mysite/urls.py]: @webknjaz what you mean ?

Comment (PetrushynskyiOleksii) [reply mysite/urls.py]: @webknjaz done

**Review by webknjaz — APPROVED** [review]

**Pull request merged**